dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/controller/ProfilingSystem.java

```
@@ -199,8 +199,13 @@ public void run() {
       } catch (final Exception e) {
         log.error("Exception in profiling thread, continuing", e);
       } catch (final Throwable t) {
-        log.error("Fatal exception in profiling thread, exiting", t);
-        throw t;
+        /*
+        Try to continue even after fatal exception. It seems to be useful to attempt to store profile when this happens.
+        For example JVM maybe out of heap and throwing OutOfMemoryError - we probably still would want to continue and
```

For OOME usually it is not really good to try and and log exception - the memory is stretched already and the logging statement can throw its own OOME and propagate the exception we are trying to mask to the caller.

I would suggest handling OOME differently and just silently ignore it.

---

good point, I'll just wrap the whole logging statement in try/catch

---

Good idea to try the log if possible!